test-data/unit/check-errorcodes.test

```diff
@@ -807,3 +807,48 @@ from typing_extensions import TypedDict
 
 Foo = TypedDict("Bar", {})  # E: First argument "Bar" to TypedDict() does not match variable name "Foo"  [name-match]
 [builtins fixtures/dict.pyi]
+[case testImplicitBool]
+# flags: --enable-error-code implicit-bool
+from typing import List, Union
+
+class Foo:
+    pass
+
+foo = Foo()
+if foo:  # E: "__main__.foo" has type "__main__.Foo" which does not implement __bool__ or __len__ so it will always be true in boolean context  [implicit-bool]
+    pass
+
+zero = 0
+if zero:
+    pass
+
+false = False
+if false:
+    pass
+
+null = None
+if null:
+    pass
+
+s = ''
+if s:
+    pass
+
+good_union: Union[str, int] = 5
+if good_union:
+    pass
+
+bad_union: Union[Foo, object] = Foo()
+if bad_union:  # E: "__main__.bad_union" has type "Union[__main__.Foo, builtins.object]" none of which implement __bool__ or __len__ so it will always be true in boolean context  [implicit-bool]
```

Comment:
```suggestion
if bad_union:  # E: "__main__.bad_union" has type "Union[__main__.Foo, builtins.object]" of which no members implement __bool__ or __len__ so it will always be true in boolean context  [implicit-bool]
```